File: 682. Baseball Game/_682.py
class Solution:
    def calPoints(self, ops) -> int:
        res = [0,0]
        for i in ops:
            if i == 'D':
                res.append(res[-1]*2)
            elif i == '+':
                res.append(res[-1]+res[-2])
            elif i == 'C':
                if len(res)>2:
                    res.pop()
            else:
                res.append(int(i))

        return sum(res)

File: 682. Baseball Game/682.go
package leetcode

import(
	"strconv"
)

func calPoints(ops []string) int {
    res := make([]int, 2)
	for _, v := range ops{
		if v == "D"{
			res = append(res, res[len(res)-1]*2) 
		}else if v == "+"{
			res = append(res, res[len(res)-1]+res[len(res)-2])
		}else if v == "C"{
			if len(res)>2{
				res = res[:len(res)-1]
			}
		}else{
			tmp, _ := strconv.Atoi(v)
			res = append(res, tmp)
		}
	}

	sum := 0
	for _,v := range res{
		sum += v
	}
	return sum
}